File: práctica 1/Punto.py
import math

class Punto:
    def _init_(self, x: float, y: float):
        self.x = x
        self.y = y
    def coord_cartesianas(self):
        return (self.x, self.y)
    def coord_polares(self):
        r = math.sqrt(self.x ** 2 + self.y ** 2)
        theta = math.atan2(self.y, self.x)  
        return (r, math.degrees(theta))  
    def _str_(self):
        return f"({self.x}, {self.y})"
    
p = Punto(9, 11)
print("Coordenadas Cartesianas:", p.coord_cartesianas())
print("Coordenadas Polares:", p.coord_polares())
print("Punto en Cadena: ",p)

File: práctica 1/Punto.java
public class Punto {
    double x, y;

    Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    double[] coord_cartesianas() {
        return new double[]{x, y};
    }

    double[] coord_polares() {
        double r = Math.sqrt(x * x + y * y);
        double t = Math.toDegrees(Math.atan2(y, x));
        return new double[]{r, t};
    }

    String toStringPunto() {
        return "Punto(" + x + ", " + y + ")";
    }

    public static void main(String[] args) {
        Punto p = new Punto(3, 4);
        System.out.println(p.toStringPunto());
        System.out.println("Cartesianas: [" + p.coord_cartesianas()[0] + ", " + p.coord_cartesianas()[1] + "]");
        System.out.println("Polares: [" + p.coord_polares()[0] + ", " + p.coord_polares()[1] + "]");
    }
}